core/trie/trie.go

```diff
@@ -68,6 +68,10 @@ func RunOnTempTrie(height uint, do func(*Trie) error) error {
 // FeltToBitSet Converts a key, given in felt, to a bitset which when followed on a [Trie],
 // leads to the corresponding [Node]
 func (t *Trie) FeltToBitSet(k *felt.Felt) *bitset.BitSet {
+	if k == (*felt.Felt)(nil) {
+		return (*bitset.BitSet)(nil)
+	}
```

Comment:
Do you think casting is necessary? Can it be the following:

```suggestion
	if k == nil {
		return nil
	}
```

Reply:
How do we end up getting a nil felt here? that should never happen

Reply:
I think @liusanchuan just passed `nil` felt for his testing purposes. I would like to make this function private in the future precisely so we don't have to do this check.